packages/@aws-cdk/aws-apigatewayv2-integrations/lib/http/private/integration.ts

```diff
@@ -63,3 +65,65 @@ export abstract class HttpPrivateIntegration implements IHttpRouteIntegration {
 
   public abstract bind(options: HttpRouteIntegrationBindOptions): HttpRouteIntegrationConfig;
 }
+
+/**
+ * Aws Service integration properties
+ *
+ * @internal
+ */
+export interface AwsServiceIntegrationProps extends CommonIntegrationProps {
+}
+
+/**
+ * The Aws Service integration resource for HTTP API
+ *
+ * @internal
+ */
+export abstract class AwsServiceIntegration implements IHttpRouteIntegration {
+
+  constructor(private readonly props: AwsServiceIntegrationProps) {
+  }
+
+  /**
+   *
+   * @internal
+   */
+  protected abstract _fulfillRole(credentialsRole: IRole): void;
+
+  /**
+   *
+   * @internal
+   */
+  protected abstract _buildRequestParameters(): { [key: string]: any };
+
+  /**
+   *
+   * @internal
+   */
+  protected abstract _integrationService(): string;
+
+  /**
+   *
+   * @internal
+   */
+  protected abstract _integrationAction(): string;
+
+  public bind(_options: HttpRouteIntegrationBindOptions): HttpRouteIntegrationConfig {
+
+    const role = new Role(_options.scope, 'Role', {
```

Comment:
should this be called ApiGatewayIntegrationRole?

Reply:
Yes. I will fix that